### ISSUE TITLE
Tests require sources to be present

### ISSUE THREAD

**Issue by tammoippen** [nest/nest-simulator#1015]
At least the two tests:

* [`regressiontests/ticket-659-copyright.py`](https://github.com/nest/nest-simulator/blob/master/testsuite/regressiontests/ticket-659-copyright.py)
* [`unittests/test_unused_names.py`](https://github.com/nest/nest-simulator/blob/master/testsuite/unittests/test_unused_names.py)

both look at the nest sources during `make installcheck`. In some cases the sources are not available during testing, e.g. packaging systems like homebrew. In the first one, if the env-var `NEST_SOURCE` is set to `SKIP`, the [test is skipped](https://github.com/nest/nest-simulator/blob/01ab7e2c1bbb5735c76a008a6c66bba358844c16/testsuite/regressiontests/ticket-659-copyright.py#L85). In the second, this is not respected. 

There should be a way to skip sources checking tests, or have them outside of `make installcheck` (aka "check the installation") as separate processes run before or after in CI (similar as static linters) .

At least for homebrew, this will hinder a formula for v2.16.0, as no patches are allowed for new formulas ([homebrew-core](https://github.com/Homebrew/homebrew-core/pull/24908#discussion_r180680447)).

**Comment by tammoippen** [comment]
The current, working version of the nest formula for homebrew-core can be seen here:

https://gist.github.com/tammoippen/8b8d708312367dfa996008cb50f56145

Only the `unittests/test_unused_names.py` test is failing. Discussions regarding nest on homebrew can be followed here (https://github.com/Homebrew/homebrew-core/pull/24908) and referenced PRs.

**Comment by jougs** [comment]
@tammoippen: you are of course totally right that the two tests you mention should really be run as part of our static code analysis or code style checks instead of during the run of the testsuite. I placed the first one there, as we did not have CI back then and this seemed like the best place to put it. I probably put the second one also there just because the first was already there and I did not think enough. Thanks a lot for mentioning this!

I will take care of this while working on fixes for #345 and #972.
